agent/pkg/singularity/singularity.go

```diff
@@ -384,6 +392,27 @@ func (s *SingularityClient) RunContainer(
 	}, nil
 }
 
+// computeImageReference computes a reference to the requested image. If an image cache
+// directory is defined and it contains the required image, a path name to the cached image
+// will be returned, else a docker reference so that the image can be downloaded.
+func (s *SingularityClient) computeImageReference(req cproto.RunSpec) string {
+	requestedImage := req.ContainerConfig.Image
```

Comment:
```suggestion
func (s *SingularityClient) computeImageReference(req cproto.RunSpec) string {
	requestedImage := req.ContainerConfig.Image
```
i might just pass the `req.ContainerConfig.Image` as `requestedImage` to begin with